src/ast/passes/codegen_llvm.cpp

```diff
@@ -653,6 +653,25 @@ ScopedExpr CodegenLLVM::visit(Builtin &builtin)
       return ScopedExpr(b_.CreateLShr(uidgid, 32));
     }
     __builtin_unreachable();
+  } else if (builtin.ident == "usermode") {
+    if (!builtin.available) {
```

Comment:
Unless we can do an actual runtime check (as part of the generated code) to support AOT. I think this check should go in an earlier pass. Like in fold_literals.cpp (or something similar) where we just transform this builtin into a literal on platforms where it's not supported.

Reply:
@jschwinger233 I think you can just, like semantic_analyser does, add a reference to the top level probe in fold_literals so you can call `check_available` e.g.
```
void SemanticAnalyser::visit(Probe &probe) {
  top_level_node_ = &probe;
```

Also, should we add a new switch for "usermode" in `check_available` or if it's the same as "reg" just add it to the conditional e.g.
```
if (func == "reg" || func == "usermode") {
```